prefect_transform/tasks.py

```diff
@@ -1,25 +1,20 @@
 """Collection of tasks to interact with Transform metrics catalog"""
-import os
 from typing import Optional, Union
 
 from prefect import task
-from transform import MQLClient
-from transform.exceptions import AuthException, QueryRuntimeException, URLException
+from transform.exceptions import QueryRuntimeException
 from transform.models import MqlMaterializeResp, MqlQueryStatusResp
 
+from prefect_transform.credentials import TransformCredentials
 from prefect_transform.exceptions import (
-    TransformAuthException,
     TransformConfigurationException,
     TransformRuntimeException,
 )
 
 
 @task
 def create_materialization(
-    api_key: Optional[str] = None,
-    api_key_env_var: Optional[str] = None,
-    mql_server_url: Optional[str] = None,
-    mql_server_url_env_var: Optional[str] = None,
+    credentials: TransformCredentials,
     materialization_name: str = None,
```

Comment:
Should this be a required arg (instead of defaulting to None)? Then we can remove the `if not ...` check below